packages/headless/src/controllers/core/facets/facet-conditions-manager/headless-facet-conditions-manager.ts

```diff
@@ -0,0 +1,194 @@
+import {CoreEngine} from '../../../../app/engine';
+import {facetOptions} from '../../../../app/reducers';
+import {
+  disableFacet,
+  enableFacet,
+} from '../../../../features/facet-options/facet-options-actions';
+import {updateFreezeCurrentValues} from '../../../../features/facets/facet-set/facet-set-actions';
+import {AnyFacetValueRequest} from '../../../../features/facets/generic/interfaces/generic-facet-request';
+import {AnyFacetValue} from '../../../../features/facets/generic/interfaces/generic-facet-response';
+import {
+  CategoryFacetSection,
+  DateFacetSection,
+  FacetOptionsSection,
+  FacetSection,
+  NumericFacetSection,
+} from '../../../../state/state-sections';
+import {loadReducerError} from '../../../../utils/errors';
+
+export interface AnyFacetValueCondition<T extends AnyFacetValueRequest> {
+  /**
+   * The `facetId` of the facet whose values are used as input by the condition.
+   */
+  parentFacetId: string;
+  /**
+   * A callback function that must evaluate to `true` for the condition to be met.
+   * @example
+   * ```ts
+   * const fileTypeHasSomeSelectedValue: AnyFacetValueCondition<FacetValueRequest> = {
+   *   parentFacetId: 'filetype',
+   *   condition: (parentValues) => parentValues.some(value => value.state === 'selected')
+   * }
+   * ```
+   * .
+   * @param parentValues - The current values of the facet whose `facetId` is the `parentFacetId` value.
+   * @returns Whether the condition is met.
+   */
+  condition(parentValues: T[]): boolean;
+}
+
+export interface FacetConditionsManagerProps {
+  /**
+   * The `facetId` of the facet to enable or disable depending on whether conditions are met.
+   */
+  facetId: string;
+  /**
+   * The conditions to evaluate.
+   *
+   * * If any of these conditions are met, the dependent facet is enabled.
```

Comment:
```suggestion
   * * If any of these conditions is met, the dependent facet is enabled.
```
(conjugates with 'any')